pallets/loans-ref/src/lib.rs

```diff
@@ -681,6 +715,50 @@ pub mod pallet {
 
 			Ok(Some(T::WeightInfo::update_portfolio_valuation(count)).into())
 		}
+
+		/// Propose a change.
+		/// The change is not performed until you call [`Pallet::modify()`].
+		#[pallet::weight(100_000_000)]
+		#[pallet::call_index(8)]
+		pub fn propose_change(
+			origin: OriginFor<T>,
+			pool_id: PoolIdOf<T>,
+			mutation: LoanChangeOf<T>,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			Self::ensure_role(pool_id, &who, PoolRole::LoanAdmin)?;
+
+			T::ChangeGuard::note(pool_id, mutation)?;
+
+			Ok(())
+		}
+
+		/// Apply a proposed change identified by a change id.
+		/// It will only perform the change if the requirements for it
+		/// are fulfilled.
+		#[pallet::weight(100_000_000)]
+		#[pallet::call_index(9)]
+		pub fn apply_change(
+			origin: OriginFor<T>,
+			pool_id: PoolIdOf<T>,
+			change_id: T::ChangeId,
+		) -> DispatchResult {
+			ensure_signed(origin)?;
+
+			let Change::Loan(loan_id, mutation) = T::ChangeGuard::released(pool_id, change_id)?;
+
+			let (_, _count) = Self::update_active_loan(pool_id, loan_id, |loan| {
+				loan.modify_with(mutation.clone())
```

Comment:
Realized we are not interchanging `change`, `modify/modification` and `mutation`. Maybe good to remove one of those synonyms?

Reply:
I understand this is as follows:
- Change: something more general, can be whatever, can be a loan mutation, a policy,... and it's more at `ChangeGuard` level.
- modify (as a verb): Action that _modifies_ an entity based on some input data.
- Mutation: _Small information unit_ that can be applied to another entity to modify it.

Maybe `Mutation` can be `Modification` instead or vice versa, and avoid one synonym.

Reply:
That actually makes sense :) I think I'd personally still change `Mutation` to `Modification` then, but I can also see the reasoning for separating it.

Reply:
And `modify_with()` to `mutate_with()` would also work for you? To leave the `Mutation` name that I think expresses better that it is _small information unit_ ready to be applied. I feel like `Modification` is too open. WDYT? Maybe I'm too nitpick 😅

Reply:
Yeah I think that's fine!